destinator.cc

```diff
@@ -412,15 +423,14 @@ destinator_trkpt_disp(const Waypoint* wpt)
   }
 
   if (wpt->creation_time.isValid()) {
-    const time_t ct = wpt->GetCreationTime().toTime_t();
-    struct tm tm = *gmtime(&ct);
-    tm.tm_mon += 1;
-    tm.tm_year -= 100;
-    int date = (tm.tm_mday * 10000) + (tm.tm_mon * 100) + tm.tm_year;
+    QDate dt = wpt->GetCreationTime().date();
+    double milliseconds = 0;
+    int date = dt.day() * 10000 + (dt.month() - 1) * 100 + (dt.year() - 1900);
     gbfputint32(date, fout);
-    double milliseconds = (tm.tm_hour * 10000) +
-      (tm.tm_min * 100) + tm.tm_sec;
-    milliseconds = (milliseconds * 1000) + (wpt->GetCreationTime().time().msec());
+
+    QTime tm = wpt->GetCreationTime().time();
+    milliseconds = tm.hour() * 10000 + tm.minute() * 100 + tm.second();
+    milliseconds = milliseconds * 1000 + tm.msec();
```

Comment:
I think this needs to use wpt->GetCreationTime()**.toUTC()**.date() and .time() to make sure of the time zone.

Reply:
Agreed.